io_scene_vrm/exporter/vrm_animation_exporter.py

```diff
@@ -0,0 +1,811 @@
+import itertools
+import struct
+from os import environ
+from pathlib import Path
+from typing import Optional
+
+import bpy
+from mathutils import Euler, Quaternion, Vector
+
+from ..common import version
+from ..common.deep import Json, make_json
+from ..common.gl import GL_FLOAT
+from ..common.gltf import pack_glb
+from ..common.logging import get_logger
+from ..common.vrm1.human_bone import (
+    HumanBoneName,
+    HumanBoneSpecification,
+    HumanBoneSpecifications,
+)
+from ..editor.vrm1 import handler as vrm1_handler
+from ..editor.vrm1.property_group import Vrm1ExpressionPropertyGroup
+
+logger = get_logger(__name__)
+
+
+class VrmAnimationExporter:
+    @staticmethod
+    def execute(
+        context: bpy.types.Context, path: Path, armature: bpy.types.Object
+    ) -> set[str]:
+        return work_in_progress(context, path, armature)
+
+
+def connect_humanoid_node_dicts(
+    human_bone_specification: HumanBoneSpecification,
+    human_bone_name_to_node_dict: dict[HumanBoneName, dict[str, Json]],
+    parent_node_dict: Optional[dict[str, Json]],
+    node_dicts: list[dict[str, Json]],
+    human_bone_name_to_node_index: dict[HumanBoneName, int],
+) -> None:
+    current_node_dict = human_bone_name_to_node_dict.get(human_bone_specification.name)
+    if isinstance(current_node_dict, dict):
+        node_index = len(node_dicts)
+        human_bone_name_to_node_index[human_bone_specification.name] = node_index
+        node_dicts.append(current_node_dict)
+        if parent_node_dict is not None:
+            children = parent_node_dict.get("children")
+            if not isinstance(children, list):
+                children = []
+                parent_node_dict["children"] = children
+            children.append(node_index)
+        parent_node_dict = current_node_dict
+    for child in human_bone_specification.children():
+        connect_humanoid_node_dicts(
+            child,
+            human_bone_name_to_node_dict,
+            parent_node_dict,
+            node_dicts,
+            human_bone_name_to_node_index,
+        )
+
+
+def work_in_progress_2(context: bpy.types.Context, armature: bpy.types.Object) -> bytes:
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this function is 97. By comparison, 99% of the functions in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the function has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this function or break it into multiple functions. For example, consider extracting the code block on lines 81-159 into a separate function.